src/build/dlopen.c

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file dlopen.c
+ * @brief Proof of concept for calling dlopen() from GNU make.
+ *
+ * @author James Jones (jejones@networkradius.com)
+ * @copyright 2020 Network RADIUS SARL (legal@networkradius.com)
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <dlfcn.h>
+#include <gnumake.h>
+
+#ifdef __APPLE__
+#  define DL_EXTENSION ".dylib"
+#elif defined (WIN32)
+#  define DL_EXTENSION ".dll"
+#else
+#  define DL_EXTENSION ".so"
+#endif
+
+char	*mk_dlopen(char const *nm, unsigned int argc, char **argv);
+int	dlopen_gmk_setup(void);
+
+int plugin_is_GPL_compatible;
+
+/** GNU make-callable layer around dlopen().
+ * 
+ * @return NULL on failure; otherwise, a string. 
+ * 
+ * @note argv[0] should point to the name of the file to open, minus the extension.
+ * 
+ */
+char * mk_dlopen(UNUSED char const *nm, UNUSED unsigned int argc, char **argv)
+{
+	void	*handle;
+	char	*extended_file_name;
+	char	*result;
+	
+	extended_file_name = gmk_alloc(1 + strlen(DL_EXTENSION) + strlen(argv[0]));
+	sprintf(extended_file_name, "%s%s", argv[0], DL_EXTENSION);
+
+	handle = dlopen(extended_file_name, RTLD_LAZY);
```

Comment:
Should be `RTLD_LAZY | RTLD_LOCAL` otherwise you pollute the global symbol table and may get symbol conflicts.